configuration/configuration.go

```diff
@@ -11,7 +11,10 @@ import (
 )
 
 // Configuration is a versioned registry configuration, intended to be provided by a yaml file, and
-// optionally modified by environment variables
+// optionally modified by environment variables.
+//
+// Note that yaml field names should never include _ characters, since this is the separator used
```

Comment:
> Added a check for this in overwriteStruct with a panic. It will cause unit tests to fail if someone makes this mistake.

On second thought, I created a dedicated unit test called TestValidateConfigStruct that walks the structure. This will make sure we detect bad tags (and field name collisions) even in child structures the other unit tests don't touch.